File: 4. Java/EX51.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX51
{
    public static void main(String[] args)
    {
        int qtd;
        double not1, not2;

        Locale.setDefault(Locale.US);

        try (Scanner ent = new Scanner(System.in))
        {
            System.out.print("# Quantos alunos serao digitados: ");
            qtd = ent.nextInt();
            ent.nextLine();
            
            double[] vetMed = new double[qtd];
            String[] vetNom = new String[qtd];

            // Lê os dados dos alunos
            for (int i = 0; i < qtd; i++)
            {
                System.out.print("\n# Digite nome: ");
                vetNom[i] = ent.nextLine();
                
                System.out.printf("# Digite 1a nota do %do aluno: ", i + 1);
                not1 = ent.nextDouble();
                
                System.out.printf("# Digite 2a nota do %do aluno: ", i + 1);
                not2 = ent.nextDouble();
                
                ent.nextLine();
                vetMed[i] = (not1 + not2) / 2; // Calcula a média
            }
            
            // Exibe alunos aprovados
            System.out.println("\n* Alunos aprovados:");
            for (int i = 0; i < qtd; i++)
            {
                if (vetMed[i] >= 6)
                {
                    System.out.println(vetNom[i]);
                }
            }
        }
    }
}

File: 6. Python/EX51.py
# Solicita a quantidade de alunos
qtd = int(input("# Quantos alunos serao digitados: "))

# Cria vetores para nomes e médias
vetNom = ["" for i in range(qtd)]  # Vetor para nomes
vetMed = [0.0 for i in range(qtd)]  # Vetor para médias

# Coleta dados dos alunos
for i in range(qtd):
    vetNom[i] = input("\n# Digite nome: ")  # Coleta o nome
    nota1 = float(input(f"# Digite 1a nota do {i+1}o aluno: "))  # Coleta a primeira nota
    nota2 = float(input(f"# Digite 2a nota do {i+1}o aluno: "))  # Coleta a segunda nota

    vetMed[i] = (nota1 + nota2) / 2  # Calcula a média

# Exibe os alunos aprovados
print("\n* Alunos aprovados: ")
for i in range(qtd):
    if vetMed[i] >= 6.0:  # Verifica se a média é maior ou igual a 6
        print(vetNom[i])  # Exibe o nome do aluno aprovado
